hal/src/main/native/athena/SPI.cpp

```diff
@@ -522,313 +516,201 @@ void HAL_SetSPIHandle(HAL_SPIPort port, int32_t handle) {
   }
 }
 
-static void spiAccumulatorProcess(uint64_t currentTime, SPIAccumulator* accum) {
-  // perform SPI transaction
-  uint8_t resp_b[4];
-  HAL_TransactionSPI(accum->port, accum->cmd, resp_b, accum->xferSize);
-
-  // convert from bytes
-  uint32_t resp = 0;
-  if (accum->bigEndian) {
-    for (int32_t i = 0; i < accum->xferSize; ++i) {
-      resp <<= 8;
-      resp |= resp_b[i] & 0xff;
-    }
-  } else {
-    for (int32_t i = accum->xferSize - 1; i >= 0; --i) {
-      resp <<= 8;
-      resp |= resp_b[i] & 0xff;
-    }
+void HAL_InitSPIAuto(HAL_SPIPort port, int32_t bufferSize, int32_t* status) {
+  if (port < 0 || port >= kSpiMaxHandles) {
+    *status = PARAMETER_OUT_OF_RANGE;
+    return;
+  }
+
+  std::lock_guard<wpi::mutex> lock(spiAutoMutex);
+  if (spiAutoPort == port) return;
```

Comment:
Should we error on an accidental auto init of the same port? In pretty much all other HAL places a double init would error out.

Reply:
The accumulator code silently accepted a duplicate init.

Reply:
Ah ok.

Reply:
what error should we give?  Just RESOURCE_IS_ALLOCATED?

Reply:
Yeah most likely.